hook.js

```diff
@@ -194,21 +267,60 @@ function createHook (meta) {
   async function getSource (url, context, parentGetSource) {
     if (hasIitm(url)) {
       const realUrl = deleteIitm(url)
-      const { imports, namespaces, setters } = await processModule({
+      const { imports, namespaces, setters: mapSetters } = await processModule({
           srcUrl: realUrl,
           context,
           parentGetSource
       })
-
+      const setters = Array.from(mapSetters.values())
+
+      // When we encounter modules that re-export all identifiers from other
+      // modules, it is possible that the transitive modules export a default
+      // identifier. Due to us having to merge all transitive modules into a
+      // single common namespace, we need to recognize these default exports
+      // and remap them to a name based on the module name. This prevents us
+      // from overriding the top-level module's (the one actually being imported
+      // by some source code) default export when we merge the namespaces.
+      const renamedDefaults = setters
+        .map(s => {
+          const matches = /let \$(.+) = (\$.+)\.default/.exec(s)
+          if (matches === null) return
+          return `_['${matches[1]}'] = ${matches[2]}.default`
+        })
+        .filter(s => s)
```

Comment:
We'll probably want to eventually rework this for performance, but for now, can you can you add a double-exclamation to make it clear that it's filtering on truthiness, and therefore eliminating `undefined` holes?
```suggestion
        .filter(s => !!s)
```

Reply:
In my opinion, the `!!` construct is very difficult to read. Would a comment suffice?